src/main/java/org/springframework/hateoas/client/Traverson.java

```diff
@@ -376,7 +404,45 @@ private String getAndFindLinkWithRel(String uri, Iterator<String> rels) {
 						responseBody));
 			}
 
-			return getAndFindLinkWithRel(link.getHref(), rels);
+			/**
+			 * Don't expand if the parameters are empty
+			 */
+			if (head.getParams().isEmpty()) {
+				return getAndFindLinkWithRel(link.getHref(), tail(rels));
+			} else {
+				Map<String, Object> combinedParams = new HashMap<String, Object>();
```

Comment:
This can be a method on `Hop`. `getMergedParameters(Map<String, Object> globalParameters)` or the like.